pkg/common/plugin/azure/msi_test.go

```diff
@@ -114,6 +109,63 @@ func TestFetchInstanceMetadata(t *testing.T) {
 	require.Equal(t, expected, metadata)
 }
 
+func TestMakeAgentID(t *testing.T) {
+	type args struct {
+		td                string
+		agentPathTemplate string
+		claims            *MSITokenClaims
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "successfully applies template",
+			args: args{
+				"example.org",
```

Comment:
can you set far names in all cases you create a new struct? it can resutl in unexpected errors if we choose to add more fields